h2o-core/src/main/java/water/Paxos.java

```diff
@@ -70,7 +70,13 @@ static synchronized int doHeartbeat( H2ONode h2o ) {
         // A new client was reported to this node so we propagate this information to all nodes in the cluster, to this
         // as well
         UDPClientEvent.ClientEvent.Type.CONNECT.broadcast(h2o);
-      } else if (H2O.ARGS.client && !H2O.isNodeInFlatfile(h2o)) {
+      } else if (!H2O.ARGS.client && h2o._heartbeat._client) {
```

Comment:
When we put the ip:port of the client node to the flatfile passed to the nodes, this node is not added to the client map and therefore it's not handled properly when it comes to disconnects. We need to ensure that client specified this way needs to be on the client map as well